packages/arb-token-bridge-ui/src/components/TransferPanel/CustomDestinationAddressConfirmationDialog.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from 'react'
+
+import { Dialog, UseDialogProps } from '../common/Dialog'
+import { ExternalLink } from '../common/ExternalLink'
+import { useNetworks } from '../../hooks/useNetworks'
+import { getNetworkName } from '../../util/networks'
+import { shortenAddress } from '../../util/CommonUtils'
+import { useArbQueryParams } from '../../hooks/useArbQueryParams'
+import { Checkbox } from '../common/Checkbox'
+
+export function CustomDestinationAddressConfirmationDialog(
+  props: UseDialogProps
+) {
+  const [{ destinationAddress = '' }] = useArbQueryParams()
+
+  const [networks] = useNetworks()
+
+  const networkName = getNetworkName(networks.destinationChain.id)
+
+  const [checkboxChecked, setCheckboxChecked] = useState(false)
+
+  function closeWithReset(confirmed: boolean) {
+    props.onClose(confirmed)
+    setCheckboxChecked(false)
+  }
+
+  return (
+    <Dialog
+      {...props}
+      title="Confirm Destination Address"
+      className="flex flex-col gap-4"
+      onClose={closeWithReset}
+      actionButtonProps={{
+        disabled: !checkboxChecked
+      }}
+    >
+      <div className="mb-4">
+        <p className="pb-2">
+          You are attempting to deposit funds to a destination address (
+          {shortenAddress(destinationAddress)}), which is the same as your
+          connected wallet address.
+          <br />
+          This is an uncommon action. If this was a mistake, please update your
+          destination address.
+        </p>
```

Comment:
I'd do this instead to make it even clearer to the user

```typescript
const connectedNetworkName = getNetworkName(networks.sourceChain.id)
```

```suggestion
        <p className="pb-2">
          You are attempting to deposit funds to the same address{' '}
          {shortenAddress(destinationAddress)} on {networkName}.
        </p>
        <p className="pb-2">
          This is an uncommon action because your smart contract wallet is only
          deployed on the currently connected chain.
        </p>
        <p className="pb-2">
          You should only send to the same address if you have control over this
          same address on both {connectedNetworkName} and {networkName}. If this
          was a mistake, please update your destination address.
        </p>
```